### ISSUE TITLE
Is it possible to get 'cv::dnn::Net' input dimensions? 

### ISSUE THREAD

**Issue by rytisss** [opencv/opencv#17903]
- OpenCV => 4.3.0:
- Operating System / Platform => Windows/Linux
- Compiler => VS2019/gcc

Is it possible to get a neural network input size (dimension)? When I load a neural network from *.pb or *pbtxt file (Tensorflow), I can't locate the information where the input dimension is defined. However when freeze and save my model in Tensorflow (as text). I can see that it saves parameters in what dimensions neural network was deployed: 

`node {
  name: "x"
  op: "Placeholder"
  attr {
    key: "dtype"
    value {
      type: DT_FLOAT
    }
  }
  attr {
    key: "shape"
    value {
      shape {
        dim {
          size: -1
        }
        dim {
          size: 128
        }
        dim {
          size: 128
        }
        dim {
          size: 1
        }
      }
    }
  }
}`

Is there any way to get a model input dimension using OpenCV DNN API?

**Comment by dkurt** [comment]
Users must know the dimensions for inputs :)

Please take a look at https://docs.opencv.org/master/d3/df0/classcv_1_1dnn_1_1Model.html - it may determine input dimensions for some of the formats.

**Comment by rytisss** [comment]
Thank you for a quick response! This actually might be a useful feature to have in some development (especially models development) cases :) maybe I will check the implementation of PROTO text parse for it :)
